x-pack/plugins/security_solution/public/timelines/components/side_panel/event_details/index.tsx

```diff
@@ -95,49 +98,38 @@ const EventDetailsPanelComponent: React.FC<EventDetailsPanelProps> = ({
     pagination,
   });
 
-  const hostRisk: HostRisk | null = data
-    ? {
-        loading: hostRiskLoading,
-        isModuleEnabled,
-        result: data,
-      }
-    : null;
+  const hostRisk: HostRisk | null = useMemo(() => {
+    return data
+      ? {
+          loading: hostRiskLoading,
+          isModuleEnabled,
+          result: data,
+        }
+      : null;
+  }, [data, hostRiskLoading, isModuleEnabled]);
 
-  if (!expandedEvent?.eventId) {
-    return null;
-  }
+  const { header, body, footer } = useMemo<{
+    header: ReactNode;
+    body: ReactNode;
+    footer: ReactNode;
+  }>(() => {
+    let bodyContent: ReactNode = null;
 
-  return isFlyoutView ? (
-    <>
-      <FlyoutHeader
-        isHostIsolationPanelOpen={isHostIsolationPanelOpen}
-        isAlert={isAlert}
-        isolateAction={isolateAction}
-        loading={loading}
-        ruleName={ruleName}
-        showAlertDetails={showAlertDetails}
-        timestamp={timestamp}
-      />
-      <FlyoutBody
-        alertId={alertId}
-        browserFields={browserFields}
-        detailsData={detailsData}
-        event={expandedEvent}
-        hostName={hostName}
-        hostRisk={hostRisk}
-        handleIsolationActionSuccess={handleIsolationActionSuccess}
-        handleOnEventClosed={handleOnEventClosed}
-        isAlert={isAlert}
-        isDraggable={isDraggable}
-        isolateAction={isolateAction}
-        isIsolateActionSuccessBannerVisible={isIsolateActionSuccessBannerVisible}
-        isHostIsolationPanelOpen={isHostIsolationPanelOpen}
-        loading={loading}
-        rawEventData={rawEventData}
-        showAlertDetails={showAlertDetails}
-        timelineId={timelineId}
-        isReadOnly={isReadOnly}
-      />
+    let headerContent: ReactNode =
+      isFlyoutView || isHostIsolationPanelOpen ? (
+        <FlyoutHeader
+          isHostIsolationPanelOpen={isHostIsolationPanelOpen}
+          isAlert={isAlert}
+          isolateAction={isolateAction}
+          loading={loading}
+          ruleName={ruleName}
+          showAlertDetails={showAlertDetails}
+          timestamp={timestamp}
+        />
+      ) : null;
+
+    // Same footer for everyone
```

Comment:
Consider reducing the complexity of this `useMemo` by moving the footer out of it, because (per the comment), it's the `Same footer for everyone`,  as [suggested here](https://github.com/elastic/kibana/pull/139881/files?diff=split&w=1#r965371500).

Reply:
Thanks for the suggestion and for your time on reviewing this. I will push through a change with this.